daemon/cmd/daemon_main.go

```diff
@@ -150,6 +158,226 @@ func skipInit(basePath string) bool {
+				"k8s-watcher-queue-size",
+				"enable-k8s-event-handover",
+				"enable-k8s-external-ips",
+				"disable-k8s-services",
+				"annotate-k8s-node",
+				"disable-endpoint-crd",
+			},
+		},
+		{
+			name: "Metrics and Monitoring related flags",
+			flags: []string{
+				"metrics",
+				"monitor-aggregation",
+				"monitor-aggregation-flags",
+				"monitor-aggregation-interval",
+				"monitor-queue-size",
+			},
+		},
+		{
+			name: "Http related flags",
+			flags: []string{
+				"http-idle-timeout",
+				"http-max-grpc-timeout",
+				"http-request-timeout",
+				"http-retry-count",
+				"http-retry-timeout",
+			},
+		},
+		{
+			name: "Logging and Trace related flags",
+			flags: []string{
+				"enable-tracing",
+				"log-driver",
+				"log-opt",
+				"log-system-load",
+				"envoy-log",
+				"prometheus-serve-addr",
+			},
+		},
+		{
+			name: "IP related flags",
+			flags: []string{
+				"enable-ipv4",
+				"enable-ipv6",
+				"ip-allocation-timeout",
+				"ipam",
+				"ipsec-key-file",
+				"ipv4-cluster-cidr-mask-size",
+				"ipv4-node",
+				"ipv4-pod-subnets",
+				"ipv4-range",
+				"ipv4-service-loopback-address",
+				"ipv4-service-range",
+				"ipv6-cluster-alloc-cidr",
+				"ipv6-node",
+				"ipv6-pod-subnets",
+				"ipv6-range",
+				"ipv6-service-range",
+				"mtu",
+				"nat46-range",
+			},
+		},
+		{
+			name: "KVstore related flags",
+			flags: []string{
+				"kvstore",
+				"kvstore-connectivity-timeout",
+				"kvstore-opt",
+				"kvstore-periodic-sync",
+			},
+		},
+		{
+			name: "Cluster related flags",
+			flags: []string{
+				"cluster-id",
+				"cluster-name",
+				"clustermesh-config",
+			},
+		},
+		{
+			name: "Route related flags",
+			flags: []string{
+				"single-cluster-route",
+				"enable-endpoint-routes",
+				"enable-local-node-route",
+				"blacklist-conflicting-routes",
+				"auto-direct-node-routes",
+			},
+		},
+		{
+			name: "Encryption related flags",
+			flags: []string{
+				"encrypt-interface",
+				"encrypt-node",
+				"enable-host-reachable-services",
+				"egress-masquerade-interfaces",
+				"enable-l7-proxy",
+				"enable-ipsec",
+			},
+		},
+		{
+			name: "Endpoint related flags",
+			flags: []string{
+				"enable-endpoint-health-checking",
+				"endpoint-interface-name-prefix",
+				"endpoint-queue-size",
+				"disable-endpoint-crd",
+			},
+		},
+		{
+			name: "Policy related flags",
+			flags: []string{
+				"allow-localhost",
+				"allow-icmp-frag-needed",
+				"enable-policy",
+				"exclude-local-address",
+				"force-local-policy-eval-at-source",
+				"host-reachable-services-protos",
+				"install-iptables-rules",
+				"policy-queue-size",
+				"prepend-iptables-chains",
+			},
+		},
+	}
+	var b bytes.Buffer
+	var seen []string // what flags have already been processed
+
+	// go through all sections defined in the config in order
+	for _, s := range sections {
+		fmt.Fprintf(&b, "%s:\n", s.name)
+		if s.desc != "" {
+			fmt.Fprintf(&b, "\n%s\n\n", s.desc)
```

Comment:
I'm wondering if it would be worth enforcing a line width. We have a few long descriptions and they look ugly right now. Not that this pull request introduced that issue, but it would be nice to have.

Reply:
Maybe we can file an issue about switching this flag printer to a `tabwriter`?